File: packages/sdk-python/src/melaya/runner.py
"""Runner API — mint, list, and revoke runner tokens.

Maps to ``/api/v1/private/runner/tokens``. Runner tokens (``mel_run_`` prefix)
authenticate the Melaya runner CLI process that executes agent pipelines on
your infrastructure.

Example
-------
>>> from melaya import Melaya
>>> m = Melaya(api_key="mk_...")
>>> result = m.runner.create_token(label="prod-server-1")
>>> token = result["token"]  # Store securely — shown only once.
>>> tokens = m.runner.list_tokens()
>>> m.runner.revoke_token(tokens[0]["id"])
"""
from __future__ import annotations

from typing import Any, Dict, List, Optional

from .platform_types import JsonDict


class RunnerAPI:
    def __init__(self, request: Any) -> None:
        self._request = request

    def create_token(self, *, label: Optional[str] = None) -> JsonDict:
        """Mint a new runner token (``mel_run_`` prefix). Plaintext shown only once — store securely."""
        body: Dict[str, Any] = {}
        if label is not None:
            body["label"] = label
        return self._request("POST", "/api/v1/private/runner/tokens", json=body if body else None)

    def list_tokens(self) -> List[JsonDict]:
        """List all runner tokens for the caller (masked, with last_seen timestamp)."""
        return self._request("GET", "/api/v1/private/runner/tokens")

    def revoke_token(self, token_id: str) -> JsonDict:
        """Revoke a runner token by ID."""
        return self._request("DELETE", f"/api/v1/private/runner/tokens/{token_id}")

File: packages/sdk-ruby/lib/melaya/runner.rb
# frozen_string_literal: true

module Melaya
  # Runner API — mint, list, and revoke runner tokens.
  #
  # Runner tokens (mel_run_ prefix) authenticate the Melaya runner CLI process
  # that executes agent pipelines on your infrastructure.
  # The plaintext token is returned only once on creation — store it securely.
  #
  # Maps to /api/v1/private/runner/tokens/*.
  #
  # @example
  #   result = melaya.runner.create_token(label: "prod-server-1")
  #   token = result["token"]   # store securely — shown only once
  #
  #   tokens = melaya.runner.list_tokens
  #   melaya.runner.revoke_token(tokens.first["id"])
  class RunnerAPI
    def initialize(http)
      @http = http
    end

    # POST /api/v1/private/runner/tokens
    # Mint a new mel_run_ runner token.
    # @param label [String, nil] human-readable label for the token
    def create_token(label: nil)
      body = label ? { "label" => label } : nil
      @http.post("/api/v1/private/runner/tokens", body)
    end

    # GET /api/v1/private/runner/tokens
    # List all runner tokens for the caller (masked, with last_seen).
    def list_tokens
      @http.get("/api/v1/private/runner/tokens")
    end

    # DELETE /api/v1/private/runner/tokens/:tokenId
    # Revoke a runner token by ID.
    # @param token_id [String]
    def revoke_token(token_id)
      @http.delete("/api/v1/private/runner/tokens/#{URI.encode_www_form_component(token_id.to_s)}")
    end
  end
end
